### ISSUE TITLE
Bug(`no-duplicate-schema-methods`): triggered on the same constraint imposed on different entities

### ISSUE THREAD

**Issue by andreww2012** [marcalexiei/eslint-zod#470]
### 📦 Affected package

eslint-plugin-zod

### 🧩 Description

The following snippet makes the rule report on allegedly duplicate `.min(N)` calls, although they impose constraints on `string` and `array` respectively.

### 🪜 How to reproduce

```ts
import * as z from 'zod';
z.string().min(1).array().min(1);
```

[ESLint playground repro](https://eslint-online-playground.netlify.app/#eNpVUMtuwyAQ/BXExUkVk1jqyTm2f1F6QGbt0JoFAa4SW/738nDdFiFgtTOzwyzUu+4Md6HtCOzD05YqbY0L5IkIT2bSO6NJNRtZXTlynJkPTuFwODKt8NAcmXBOPPbySk8U/KgwsM5gr4Z/mguR0CuEl9wia1HnG+NcGJzGSRshzt0cFEhtx2lQWO9+4J5xUVZMY753+cNbBIQlHTwUnm9JrEleUeJU3mu8E8ZNIxREqnhIvz6jqeVkR9WJALXvbqBFrSHcjPRVG205Z1y1CRShdL7nIKzoPsWQYjUYQ8ijOQ0PC5y28aWNjCM5zT44lfD1ChZQAnYKfMJsbn8SKrTmwpqGXTbeb/dPOAX4zJq4OU2wleNK12/9s6cj)

### ✅ Expected behavior

The rule is not triggered.

### 💻 System info

```shell
Binaries:
    Node: 22.23.1 - /Users/<...>/.nvm/versions/node/v22.23.1/bin/node
    npm: 10.9.8 - /Users/<...>/.nvm/versions/node/v22.23.1/bin/npm
    pnpm: 11.26.0 - /Users/<...>/.nvm/versions/node/v22.23.1/bin/pnpm
  npmPackages:
    eslint: 10.10.0 => 10.10.0 
    eslint-plugin-zod: 4.14.1 => 4.14.1
```

### 🙋 Contribution

- [ ] I am willing to open a pull request to fix this
